### PR TITLE
615 fuji runner external file

### PR DESCRIPTION
**Pull request by kpapag**
change-log

- Added support for assessment using FAIR EVA app
- Added support for file logging system
- Added support for timeout - retry for all API Calls
- Added support for file input of one or more sitemaps to assess
- Added support for file input of one or more studies (same or different SP) to assess
- Added support for assessing studies of CDC Staging Env (currently only F-UJI API)
- Added support for saving failed studies to a file and re-assess them before script ends
- Added support for inputting Service Provider sitemaps beside CESSDA (previous version only run on CDC sitemap by default)
- Added (studyNumber, Publisher) to CSV for scores comparison, obtained by CDC Internal AP

### CODE REVIEW

File: src/sitemaps.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  logger.info(`Processing sitemap: ${sitemapLine}`);
+  dashLogger.info(`Processing sitemap: ${sitemapLine}, time:${new Date().toUTCString()}`);
+  const studiesAssessFiltered: string[] = await getStudiesFromSitemap(new URL(sitemapLine));
+  const outputName: string = new URL(sitemapLine).hostname;
+  await getStudiesAssess(studiesAssessFiltered, outputName);
+  logger.info(`Finished assessing sitemap: ${sitemapLine}`);
+  dashLogger.info(`Finished assessing sitemap: ${sitemapLine}, time:${new Date().toUTCString()}`);
+}
+//check file if any studies failed and re-assess them
+await isFileEmpty('../outputs/failed.txt')
```
Comment (matthew-morris-cessda) [inline src/sitemaps.ts:32]: `../outputs/failed.txt` is removed at line 16, so this will never work as intended

File: src/types/global.d.ts
Comment (matthew-morris-cessda) [inline src/types/global.d.ts:1]: It would be better to simply export `StudyInfo` directly rather than using the global namespace

**Review by matthew-morris-cessda — CHANGES_REQUESTED** [review]
I've reviewed the changed code and left some comments and suggested changes

**Review by matthew-morris-cessda — APPROVED** [review]

**Comment by matthew-morris-cessda** [comment]
@john-shepherdson Could you take a look?

**Review by john-shepherdson — APPROVED** [review]

**Pull request merged**